[BUG]delta_lake_update_test.py::test_delta_update_partitions[['a', 'b']-False] failed by DATAGEN_SEED=1707683137

**Describe the bug**
delta_lake_update_test.py::test_delta_update_partitions[['a', 'b']-False][DATAGEN_SEED=1707683137

```
[2024-02-11T20:47:42.987Z] =================================== FAILURES ===================================

[2024-02-11T20:47:42.987Z] ________________ test_delta_update_partitions[['a', 'b']-False] ________________

[2024-02-11T20:47:42.987Z] [gw4] linux -- Python 3.9.18 /opt/conda/bin/python

[2024-02-11T20:47:42.987Z] 

[2024-02-11T20:47:42.987Z] spark_tmp_path = '/tmp/pyspark_tests//it-test-331-663-4hk89-v63bc-gw4-1515130-1140402737/'

[2024-02-11T20:47:42.987Z] use_cdf = False, partition_columns = ['a', 'b']

[2024-02-11T20:47:42.987Z] 

[2024-02-11T20:47:42.987Z]     @allow_non_gpu(*delta_meta_allow)

[2024-02-11T20:47:42.987Z]     @delta_lake

[2024-02-11T20:47:42.987Z]     @ignore_order

[2024-02-11T20:47:42.987Z]     @pytest.mark.parametrize("use_cdf", [True, False], ids=idfn)

[2024-02-11T20:47:42.988Z]     @pytest.mark.parametrize("partition_columns", [["a"], ["a", "b"]], ids=idfn)

[2024-02-11T20:47:42.988Z]     @pytest.mark.skipif(is_before_spark_320(), reason="Delta Lake writes are not supported before Spark 3.2.x")

[2024-02-11T20:47:42.988Z]     def test_delta_update_partitions(spark_tmp_path, use_cdf, partition_columns):

[2024-02-11T20:47:42.988Z]         def generate_dest_data(spark):

[2024-02-11T20:47:42.988Z]             return three_col_df(spark,

[2024-02-11T20:47:42.988Z]                                 SetValuesGen(IntegerType(), range(5)),

[2024-02-11T20:47:42.988Z]                                 SetValuesGen(StringType(), "abcdefg"),

[2024-02-11T20:47:42.988Z]                                 string_gen)

[2024-02-11T20:47:42.988Z]         update_sql = "UPDATE delta.`{path}` SET a = 3 WHERE b < 'c'"

[2024-02-11T20:47:42.988Z] >       assert_delta_sql_update_collect(spark_tmp_path, use_cdf, generate_dest_data,

[2024-02-11T20:47:42.988Z]                                         update_sql, partition_columns)

[2024-02-11T20:47:42.988Z] 

[2024-02-11T20:47:42.988Z] ../../src/main/python/delta_lake_update_test.py:116: 

[2024-02-11T20:47:42.988Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-02-11T20:47:42.988Z] ../../src/main/python/delta_lake_update_test.py:61: in assert_delta_sql_update_collect

[2024-02-11T20:47:42.988Z]     delta_sql_update_test(spark_tmp_path, use_cdf, dest_table_func, update_sql, checker,

[2024-02-11T20:47:42.988Z] ../../src/main/python/delta_lake_update_test.py:36: in delta_sql_update_test

[2024-02-11T20:47:42.988Z]     check_func(data_path, do_update)

[2024-02-11T20:47:42.988Z] ../../src/main/python/delta_lake_update_test.py:60: in checker

[2024-02-11T20:47:42.988Z]     with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))

[2024-02-11T20:47:42.988Z] ../../src/main/python/spark_session.py:147: in with_cpu_session

[2024-02-11T20:47:42.988Z]     return with_spark_session(func, conf=copy)

[2024-02-11T20:47:42.988Z] /opt/conda/lib/python3.9/contextlib.py:79: in inner

[2024-02-11T20:47:42.988Z]     return func(*args, **kwds)

[2024-02-11T20:47:42.988Z] ../../src/main/python/spark_session.py:131: in with_spark_session

[2024-02-11T20:47:42.988Z]     ret = func(_spark)

[2024-02-11T20:47:42.988Z] ../../src/main/python/delta_lake_update_test.py:60: in <lambda>

[2024-02-11T20:47:42.988Z]     with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))

[2024-02-11T20:47:42.988Z] ../../src/main/python/delta_lake_utils.py:139: in assert_gpu_and_cpu_delta_logs_equivalent

[2024-02-11T20:47:42.988Z]     assert_delta_log_json_equivalent(file, cpu_json, gpu_json)

[2024-02-11T20:47:42.988Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2024-02-11T20:47:42.988Z] 

[2024-02-11T20:47:42.988Z] filename = '00000000000000000001.json'

[2024-02-11T20:47:42.988Z] c_json = {'add': {'dataChange': True, 'partitionValues': {'a': '3', 'b': 'b'}, 'path': 'a=3/b=b/partsnappy.parquet', 'stats': '...x82ÛÂrº\\u000F\xad"},"maxValues":{"c":"ÿûÜÌ\x9bã±½?«¨\\u00171}mò\\u0015>Ò\\u0011f\x9a¤óRúTÄÂ0"},"nullCount":{"c":7}}'}}

[2024-02-11T20:47:42.988Z] g_json = {'add': {'dataChange': True, 'partitionValues': {'a': '3', 'b': 'b'}, 'path': 'a=3/b=b/partsnappy.parquet', 'stats': '...x82ÛÂrº\\u000F\xad"},"maxValues":{"c":"ÿûÜÌ\x9bã±½?«¨\\u00171}mò\\u0015>Ò\\u0011f\x9a¤óRúTÄÂ0"},"nullCount":{"c":9}}'}}

[2024-02-11T20:47:42.988Z] 

[2024-02-11T20:47:42.988Z]     def assert_delta_log_json_equivalent(filename, c_json, g_json):

[2024-02-11T20:47:42.988Z]         assert c_json.keys() == g_json.keys(), "Delta log {} has mismatched keys:\nCPU: {}\nGPU: {}".format(filename, c_json, g_json)

[2024-02-11T20:47:42.988Z]         def fixup_path(d):

[2024-02-11T20:47:42.988Z]             """Modify the 'path' value to remove random IDs in the pathname"""

[2024-02-11T20:47:42.988Z]             parts = d["path"].split("-")

[2024-02-11T20:47:42.988Z]             d["path"] = "-".join(parts[0:1]) + ".".join(parts[-1].split(".")[-2:])

[2024-02-11T20:47:42.988Z]         def del_keys(key_list, c_val, g_val):

[2024-02-11T20:47:42.988Z]             for key in key_list:

[2024-02-11T20:47:42.988Z]                 c_val.pop(key, None)

[2024-02-11T20:47:42.988Z]                 g_val.pop(key, None)

[2024-02-11T20:47:42.988Z]         for key, c_val in c_json.items():

[2024-02-11T20:47:42.989Z]             g_val = g_json[key]

[2024-02-11T20:47:42.989Z]             # Strip out the values that are expected to be different

[2024-02-11T20:47:42.989Z]             c_tags = c_val.get("tags", {})

[2024-02-11T20:47:42.989Z]             g_tags = g_val.get("tags", {})

[2024-02-11T20:47:42.989Z]             del_keys(["INSERTION_TIME", "MAX_INSERTION_TIME", "MIN_INSERTION_TIME"], c_tags, g_tags)

[2024-02-11T20:47:42.989Z]             if key == "metaData":

[2024-02-11T20:47:42.989Z]                 assert c_val.keys() == g_val.keys(), "Delta log {} 'metaData' keys mismatch:\nCPU: {}\nGPU: {}".format(filename, c_val, g_val)

[2024-02-11T20:47:42.989Z]                 del_keys(("createdTime", "id"), c_val, g_val)

[2024-02-11T20:47:42.989Z]             elif key == "add":

[2024-02-11T20:47:42.989Z]                 assert c_val.keys() == g_val.keys(), "Delta log {} 'add' keys mismatch:\nCPU: {}\nGPU: {}".format(filename, c_val, g_val)

[2024-02-11T20:47:42.989Z]                 del_keys(("modificationTime", "size"), c_val, g_val)

[2024-02-11T20:47:42.989Z]                 fixup_path(c_val)

[2024-02-11T20:47:42.989Z]                 fixup_path(g_val)

[2024-02-11T20:47:42.989Z]             elif key == "cdc":

[2024-02-11T20:47:42.989Z]                 assert c_val.keys() == g_val.keys(), "Delta log {} 'cdc' keys mismatch:\nCPU: {}\nGPU: {}".format(filename, c_val, g_val)

[2024-02-11T20:47:42.989Z]                 del_keys(("size",), c_val, g_val)

[2024-02-11T20:47:42.989Z]                 fixup_path(c_val)

[2024-02-11T20:47:42.989Z]                 fixup_path(g_val)

[2024-02-11T20:47:42.989Z]             elif key == "commitInfo":

[2024-02-11T20:47:42.989Z]                 assert c_val.keys() == g_val.keys(), "Delta log {} 'commitInfo' keys mismatch:\nCPU: {}\nGPU: {}".format(filename, c_val, g_val)

[2024-02-11T20:47:42.989Z]                 del_keys(("timestamp", "txnId"), c_val, g_val)

[2024-02-11T20:47:42.989Z]                 for v in c_val, g_val:

[2024-02-11T20:47:42.989Z]                     _fixup_operation_metrics(v.get("operationMetrics", {}))

[2024-02-11T20:47:42.989Z]                     _fixup_operation_parameters(v.get("operationParameters", {}))

[2024-02-11T20:47:42.989Z]             elif key == "remove":

[2024-02-11T20:47:42.989Z]                 assert c_val.keys() == g_val.keys(), "Delta log {} 'remove' keys mismatch:\nCPU: {}\nGPU: {}".format(filename, c_val, g_val)

[2024-02-11T20:47:42.989Z]                 del_keys(("deletionTimestamp", "size"), c_val, g_val)

[2024-02-11T20:47:42.989Z]                 fixup_path(c_val)

[2024-02-11T20:47:42.989Z]                 fixup_path(g_val)

[2024-02-11T20:47:42.989Z] >           assert c_val == g_val, "Delta log {} is different at key '{}':\nCPU: {}\nGPU: {}".format(filename, key, c_val, g_val)

[2024-02-11T20:47:42.989Z] E           AssertionError: Delta log 00000000000000000001.json is different at key 'add':

[2024-02-11T20:47:42.989Z] E           CPU: {'path': 'a=3/b=b/partsnappy.parquet', 'partitionValues': {'a': '3', 'b': 'b'}, 'dataChange': True, 'stats': '{"numRecords":147,"minValues":{"c":"\\u0001\x80\\\\\\f\\u00039\x9eÜVJ³¾\x93÷æô¸è=¯ûo`\x82ÛÂrº\\u000F\xad"},"maxValues":{"c":"ÿûÜÌ\x9bã±½?«¨\\u00171}mò\\u0015>Ò\\u0011f\x9a¤óRúTÄÂ0"},"nullCount":{"c":7}}'}

[2024-02-11T20:47:42.989Z] E           GPU: {'path': 'a=3/b=b/partsnappy.parquet', 'partitionValues': {'a': '3', 'b': 'b'}, 'dataChange': True, 'stats': '{"numRecords":149,"minValues":{"c":"\\u0001\x80\\\\\\f\\u00039\x9eÜVJ³¾\x93÷æô¸è=¯ûo`\x82ÛÂrº\\u000F\xad"},"maxValues":{"c":"ÿûÜÌ\x9bã±½?«¨\\u00171}mò\\u0015>Ò\\u0011f\x9a¤óRúTÄÂ0"},"nullCount":{"c":9}}'}

[2024-02-11T20:47:42.989Z] 

[2024-02-11T20:47:42.989Z] ../../src/main/python/delta_lake_utils.py:104: AssertionError
```

**Steps/Code to reproduce bug**
Run IT on Spark 3.3.1 with datagen_seed=1707683137. 

**Expected behavior**
Test case should succeed.

--- a/integration_tests/src/main/python/delta_lake_delete_test.py
+++ b/integration_tests/src/main/python/delta_lake_delete_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2023, NVIDIA CORPORATION.
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -55,7 +55,10 @@ def assert_delta_sql_delete_collect(spark_tmp_path, use_cdf, dest_table_func, de
         cpu_result = with_cpu_session(lambda spark: read_data(spark, cpu_path).collect(), conf=conf)
         gpu_result = with_cpu_session(lambda spark: read_data(spark, gpu_path).collect(), conf=conf)
         assert_equal(cpu_result, gpu_result)
-        with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
+        # Using partition columns involves sorting, and there's no guarantees on the task
+        # partitioning due to random sampling.
+        if not partition_columns:
+            with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
     delta_sql_delete_test(spark_tmp_path, use_cdf, dest_table_func, delete_sql, checker,
                           partition_columns)
 

--- a/integration_tests/src/main/python/delta_lake_merge_test.py
+++ b/integration_tests/src/main/python/delta_lake_merge_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2023, NVIDIA CORPORATION.
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -68,7 +68,9 @@ def assert_delta_sql_merge_collect(spark_tmp_path, spark_tmp_table_factory, use_
         cpu_result = with_cpu_session(lambda spark: read_data(spark, cpu_path).collect(), conf=conf)
         gpu_result = with_cpu_session(lambda spark: read_data(spark, gpu_path).collect(), conf=conf)
         assert_equal(cpu_result, gpu_result)
-        if compare_logs:
+        # Using partition columns involves sorting, and there's no guarantees on the task
+        # partitioning due to random sampling.
+        if compare_logs and not partition_columns:
             with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
     delta_sql_merge_test(spark_tmp_path, spark_tmp_table_factory, use_cdf,
                          src_table_func, dest_table_func, merge_sql, checker, partition_columns)

--- a/integration_tests/src/main/python/delta_lake_update_test.py
+++ b/integration_tests/src/main/python/delta_lake_update_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2023, NVIDIA CORPORATION.
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -55,8 +55,9 @@ def assert_delta_sql_update_collect(spark_tmp_path, use_cdf, dest_table_func, up
         gpu_result = with_cpu_session(lambda spark: read_data(spark, gpu_path).collect(), conf=conf)
         assert_equal(cpu_result, gpu_result)
         # Databricks not guaranteed to write the same number of files due to optimized write when
-        # using partitions
-        if not is_databricks_runtime() or not partition_columns:
+        # using partitions. Using partition columns involves sorting, and there's no guarantees on
+        # the task partitioning due to random sampling.
+        if not is_databricks_runtime() and not partition_columns:
             with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
     delta_sql_update_test(spark_tmp_path, use_cdf, dest_table_func, update_sql, checker,
                           partition_columns, enable_deletion_vectors)

--- a/integration_tests/src/main/python/delta_lake_write_test.py
+++ b/integration_tests/src/main/python/delta_lake_write_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+# Copyright (c) 2022-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -115,11 +115,8 @@ def test_delta_part_write_round_trip_unmanaged(spark_tmp_path, gens):
         lambda spark, path: spark.read.format("delta").load(path),
         data_path,
         conf=copy_and_update(writer_confs, delta_writes_enabled_conf))
-    # Databricks will sometimes generate tons of tiny files on the CPU when using floating point
-    # partition keys. The GPU does not, and this triggers a delta log mismatch. Data contents
-    # of the table are correct, and this seems like Databricks bug on the CPU.
-    if not (is_databricks_runtime() and gens.data_type in (FloatType(), DoubleType())):
-        with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
+    # Avoid checking delta log equivalence here. Using partition columns involves sorting, and
+    # there's no guarantees on the task partitioning due to random sampling.
 
 @allow_non_gpu(*delta_meta_allow)
 @delta_lake
@@ -136,11 +133,8 @@ def test_delta_multi_part_write_round_trip_unmanaged(spark_tmp_path, gens):
         lambda spark, path: spark.read.format("delta").load(path).filter("c='x'"),
         data_path,
         conf=copy_and_update(writer_confs, delta_writes_enabled_conf))
-    # Databricks will sometimes generate tons of tiny files on the CPU when using floating point
-    # partition keys. The GPU does not, and this triggers a delta log mismatch. Data contents
-    # of the table are correct, and this seems like Databricks bug on the CPU.
-    if not (is_databricks_runtime() and gens.data_type in (FloatType(), DoubleType())):
-        with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
+    # Avoid checking delta log equivalence here. Using partition columns involves sorting, and
+    # there's no guarantees on the task partitioning due to random sampling.
 
 def do_update_round_trip_managed(spark_tmp_path, mode):
     gen_list = [("x", int_gen), ("y", binary_gen), ("z", string_gen)]
@@ -318,7 +312,6 @@ def test_delta_overwrite_dynamic_missing_clauses(spark_tmp_table_factory, spark_
     _assert_sql(data_path, confs, "INSERT INTO delta.`{path}` VALUES (2L, 'dummy'), (4L, 'value')")
     _assert_sql(data_path, confs, "INSERT OVERWRITE TABLE delta.`{path}` " +
                 f"{clause} SELECT * FROM {view}")
-    with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
 
 @allow_non_gpu(*delta_meta_allow)
 @delta_lake
@@ -342,7 +335,8 @@ def test_delta_overwrite_mixed_clause(spark_tmp_table_factory, spark_tmp_path, m
     _assert_sql(data_path, confs, "INSERT INTO delta.`{path}` VALUES (2L, 'dummy', 23), (4L, 'value', 2)")
     _assert_sql(data_path, confs, "INSERT OVERWRITE TABLE delta.`{path}` " +
                 f"{clause} SELECT * FROM {view}")
-    with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
+    # Avoid checking delta log equivalence here. Using partition columns involves sorting, and
+    # there's no guarantees on the task partitioning due to random sampling.
 
 @allow_non_gpu(*delta_meta_allow)
 @delta_lake
@@ -1004,7 +998,9 @@ def test_delta_write_partial_overwrite_replace_where(spark_tmp_path):
         lambda spark, path: spark.read.format("delta").load(path),
         data_path,
         conf=copy_and_update(writer_confs, delta_writes_enabled_conf))
-    with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
+    # Avoid checking delta log equivalence here. Using partition columns involves sorting, and
+    # there's no guarantees on the task partitioning due to random sampling.
+    #
     # overwrite with a subset of the original schema
     gen_list = [("b", SetValuesGen(StringType(), ["y"])),
                 ("e", long_gen),
@@ -1019,7 +1015,8 @@ def test_delta_write_partial_overwrite_replace_where(spark_tmp_path):
         lambda spark, path: spark.read.format("delta").load(path),
         data_path,
         conf=copy_and_update(writer_confs, delta_writes_enabled_conf))
-    with_cpu_session(lambda spark: assert_gpu_and_cpu_delta_logs_equivalent(spark, data_path))
+    # Avoid checking delta log equivalence here. Using partition columns involves sorting, and
+    # there's no guarantees on the task partitioning due to random sampling.
 
 # ID mapping is supported starting in Delta Lake 2.2, but currently cannot distinguish
 # Delta Lake 2.1 from 2.2 in tests. https://github.com/NVIDIA/spark-rapids/issues/9276

